sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts

```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
+          LEGAL_SIGNING_AUTHORITY_MSG,
+        );
+      }
+    }
+
+    // remove existing associations and add new associations
+    await this.institutionService.updateInstitutionUser(
+      payload,
+      institutionUser,
+    );
+  }
+
+  /**
+   * Update the active status of the user.
+   * @param token
+   * @param userName
+   * @param body
+   */
+  @ApiNotFoundResponse({
+    description: "User to be updated not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be updated doesn't belong to institution of logged in user.",
+  })
+  @IsInstitutionAdmin()
+  @Patch("user-status/:userName")
+  async updateUserStatus(
+    @UserToken() token: IInstitutionUserToken,
+    @Param("userName") userName: string,
+    @Body() body: UserActiveStatusAPIInDTO,
+  ): Promise<void> {
+    // Check  user exists or not
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(userName);
+
+    if (!institutionUser) {
+      throw new NotFoundException("Institution user to be updated not found.");
+    }
+
+    // checking if user belong to logged-in users institution
+    if (institutionUser.institution.id !== token.authorizations.institutionId) {
+      throw new UnprocessableEntityException(
+        "user does not belong to your institution.",
+      );
+    }
+    await this.userService.updateUserStatus(
+      institutionUser.user.id,
+      body.isActive,
+    );
+  }
+
+  /**
+   * Get details of user who is logged in.
+   * @param token
+   * @returns User details.
+   */
+  @Get("my-details")
+  async getMyDetails(
+    @UserToken() token: IInstitutionUserToken,
+  ): Promise<InstitutionUserDetailAPIOutDTO> {
+    // Get logged in user and location details with auth
+    const userDetails = await this.userService.getUser(token.userName);
+    const user = {
+      firstName: userDetails?.firstName,
+      lastName: userDetails?.lastName,
+      isActive: userDetails?.isActive,
+      isAdmin: token.authorizations.isAdmin(),
+      email: userDetails?.email,
+    };
+
+    const authorizations = {
+      institutionId: token.authorizations.institutionId,
+      authorizations: token.authorizations.authorizations.map((el) => {
+        return {
+          locationId: el.locationId,
+          userType: el.userType,
+          userRole: el.userRole,
+        };
+      }),
+    };
+    return {
+      user: user,
+      authorizations: authorizations,
+    };
+  }
+
+  /**
+   * Get Location details of logged in user.
+   * @param userToken
+   * @returns Location details.
+   */
+  @Get("my-locations")
+  async getMyInstitutionLocations(
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<InstitutionUserLocationsAPIOutDTO[]> {
+    // get all institution locations that user has access too.
+    const InstitutionLocationData = userToken.authorizations.isAdmin()
+      ? await this.locationService.getAllInstitutionLocations(
```

Comment:
I believe that we no longer need this condition. In the past `.getLocationsIds()` was not loaded for admins users, but they are now supposed to be present due to the authorization. Hence the call `getMyInstitutionLocations` should work for both types of users.
If not part of the PR please add a TODO.